imagescan/delta.go

```diff
@@ -351,35 +349,41 @@ func getContainerRuntime(containerID string) imgcollectorconfig.Runtime {
 	return ""
 }
 
-func getPodOwnerID(pod *corev1.Pod, rsMap map[string]*appsv1.ReplicaSet, jobsMap map[string]*batchv1.Job) string {
+func (d *deltaState) getPodOwnerID(pod *corev1.Pod) string {
 	if len(pod.OwnerReferences) == 0 {
 		return string(pod.UID)
 	}
-
 	ref := pod.OwnerReferences[0]
 
 	switch ref.Kind {
 	case "ReplicaSet":
-		for _, val := range rsMap {
-			if val.UID == ref.UID {
-				if len(val.OwnerReferences) > 0 {
-					return string(val.OwnerReferences[0].UID)
-				}
-				return string(ref.UID)
-			}
+		rs, found := d.replicaSets[ref.UID]
+		if found {
+			return string(d.findNextOwnerID(rs, "Deployment"))
 		}
 	case "Job":
-		for _, val := range jobsMap {
-			if val.UID == ref.UID {
-				if len(val.OwnerReferences) > 0 {
-					return string(val.OwnerReferences[0].UID)
-				}
-				return string(ref.UID)
-			}
+		job, found := d.jobs[ref.UID]
+		if found {
+			return string(d.findNextOwnerID(job, "CronJob"))
+		}
+	}
+
+	return string(pod.UID)
+}
+
+func (d *deltaState) findNextOwnerID(obj kube.Object, expectedKind string) types.UID {
+	refs := obj.GetOwnerReferences()
+	if len(refs) == 0 {
```

Comment:
So now we will always send owner ID ?

Reply:
This is ReplicaSet and Job. Standalone pod will have owner ids as their uid. We always send owner. Owner here is found for image and image not for pod.